go/multipooler/manager/manager_test.go

```diff
@@ -470,3 +470,107 @@ func TestGetBackupLocation(t *testing.T) {
 	// Test accessing backup location field
 	assert.Equal(t, "/var/backups/pgbackrest", manager.backupLocation)
 }
+
+// TestWaitUntilReady_Success verifies that WaitUntilReady returns immediately
+// when the manager is already in Ready state
+func TestWaitUntilReady_Success(t *testing.T) {
+	logger := slog.Default()
+	config := &Config{
+		ConsensusEnabled: false,
+	}
+
+	pm := NewMultiPoolerManagerWithTimeout(logger, config, 100*time.Millisecond)
+
+	// Simulate immediate ready state
+	pm.mu.Lock()
+	pm.state = ManagerStateReady
+	pm.topoLoaded = true
+	pm.mu.Unlock()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	err := pm.WaitUntilReady(ctx)
```

Comment:
Feels like this would be a stronger test if the `WaitUntilReady()` really had to do some waiting? Maybe `WaitUntilReady` and require.NoError() could be called in a separate goroutine via waitgroup prior changing the state?

Similar comment for other tests